ladybug/src/main/java/nl/nn/ibistesttool/filter/GrayBox.java

```diff
@@ -30,36 +33,49 @@ public boolean match(Report report, Checkpoint checkpoint) {
 		if (checkpoint.getType() == Checkpoint.TYPE_INPUTPOINT || checkpoint.getType() == Checkpoint.TYPE_OUTPUTPOINT
 				 || checkpoint.getType() == Checkpoint.TYPE_INFOPOINT) {
 			List<Checkpoint> checkpoints = report.getCheckpoints();
-			for (int i = checkpoints.indexOf(checkpoint) - 1; i > 1; i--) {
-				if (checkpoints.get(i).getType() == Checkpoint.TYPE_STARTPOINT) {
-					return match(report, checkpoints.get(i));
+			ListIterator<Checkpoint> iterator = report.getCheckpoints().listIterator(checkpoints.indexOf(checkpoint));
```

Comment:
Als ik het goed lees, dan worden de eerste twee elementen niet gecheckt.

Ik weet niet waarom niet, misschien een goed idee om dat in een kort commentaar toe te lichten?

Reply:
Mijn idee hierbij was dat je de eerste checkpoint nooit wil verbergen maar eigenlijk zorgt deze "skip" daar helemaal niet voor (we zijn hier voor een ander checkpoint aan het controleren of die getoond moet worden). Je zou nog kunnen zeggen dat het eerste checkpoint nooit een Sender zal zijn en je hem hem daarom zou kunnen skippen maar dan hou ik eigenlijk liever de code wat simpeler en maar er gewoon iterator.hasPrevious() van (heb ik nu gedaan). Overigens zie ik soms wel reports die met een Sender beginnen dus daarom misschien ook wel netjes om iterator.hasPrevious() te doen.